Automattic/pocket-casts-ios
# Setting Sync: General settings

| 📘 Part of: #1400 | Depends on #1423 |
|:---:|:---:|

Sync all settings in the General screen.

<img width=300 src="https://github.com/Automattic/pocket-casts-ios/assets/3250/89ae791d-96bf-4f77-9be6-7e2283543027">

## Changes

I've mostly kept each setting to its own commit, except in cases where settings are in logical groups or pairs.

> [!NOTE]
> I chose to keep the old settings writing to their existing User Defaults. I realized it makes more sense to continue saving these as long as we have a Feature Flag in place just in case it's disabled. Otherwise, a whole other migration step is needed to go from `AppSettings` back to the old user defaults.

## To test

I won't make an exhaustive list of which settings to toggle, but all settings on this screen should sync.

#### Basic Sync
* Install the app on two devices/simulators
* Log in to the same account on each device
* Change settings on one device
* Back out to "Profile", and pull to refresh
* Pull to refresh in "Profile" on the second device
* Check that the second device's settings are updated

#### Merging Sync
* Install the app on two devices/simulators
* Log in to the same account on each device
* Change settings on one device
* Change different settings on the second device
* Back out to "Profile", and pull to refresh
* Pull to refresh in "Profile" on the second device
* Check that the second device's settings are updated with the ones changed from first as well as its' own
* Check that the first device's settings are updated with the ones changed from second as well as its' own

## Checklist
